### PR TITLE
Feature/add engagment

### LINKED ISSUE #23: Add creating engagment view

### PR DESCRIPTION
**Pull request by KonradMiskiewicz**
Resolves #23

### CODE REVIEW

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+/**
+ * Class responsible for adding engagments
+ * Parsing data to JSON and sending to the server
+ */
+public class EngagmentAdd extends AppCompatActivity implements View.OnClickListener {
+    private EditText advertisment_title_input;
+    private EditText advertisment_description_input;
+
+    private Spinner advertisment_city;
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: Those enters are redundant.

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+            @Override
+              public void onErrorResponse(VolleyError error) {
+                ErrorTextView.setText(error.getMessage());
+                Log.e(TAG, "Adding Error: " + error.getMessage());
+              }
+        })
+        {
+            public Map<String, String> getHeaders() throws AuthFailureError {
+
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: Unnecessary enter

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    /**
+     * Makes request to server and add engagment if success or shows
+     * error when failure.
+     */
+    public void onClick(View v)
+    {
+        JSONObject data = this.addParams();
+        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest
+        (Request.Method.POST, Engagment_add_URL, data, new Response.Listener<JSONObject>() {
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: Indent pls and `(` should go higher

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+            }
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+            }
+        });
+        sign_Advertisment = findViewById(R.id.Advertiser_Add);
+        sign_Advertisment.setOnClickListener(this);
+    }
+
+    @Override
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java:74]: No break between method and override

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        advertisment_title_input = findViewById(R.id.Advertisment_Title);
+        advertisment_description_input = findViewById(R.id.Advertiser_Description);
+        advertisment_city = findViewById(R.id.Advertiser_City);
+        ErrorTextView = findViewById(R.id.textView5);
+        Cities = new ArrayList<String>();
+        Cities.add("Wybierz miasto");
+        Cities.add("Łódź");
+        Cities.add("Poznań");
+        Cities.add("Wrocław");
+        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(EngagmentAdd.this,android.R.layout.simple_dropdown_item_1line,Cities);
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: https://www.youtube.com/watch?v=6_trCfsrw3U + spaces after commas

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+        // Adding request to request queue
+        VolleyInstance.getInstance(this).addToRequestQueue(jsonObjectRequest, "Create Engagment");
+    }
+
+    /**
+     * Method is responsibility for adding params to engagment
+     * @return CreateJson object
+     */
+    public JSONObject addParams() {
+
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: Remove enter

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        String advertiser_title = advertisment_title_input.getText().toString();
+        String advertisment_description = advertisment_description_input.getText().toString();
+
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: Remove at least one enter

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+        cj.addStr("city", selected_city);
+        cj.addStr("title" , advertiser_title);
+        cj.addStr("description", advertisment_description);
+
+        try {
+            return cj.makeJSON();
+        }
+        catch (JSONException e) {
+            e.printStackTrace();
+        }
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java:129]: enter here

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+     */
+    public void onClick(View v)
+    {
+        JSONObject data = this.addParams();
+        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest
+        (Request.Method.POST, Engagment_add_URL, data, new Response.Listener<JSONObject>() {
+
+            @Override
+            public void onResponse(JSONObject response) {
+
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: no enter

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+    private Button sign_Advertisment;
+
+    private TextView ErrorTextView;
+
+    List<String> Cities ;
+
+    String selected_city;
+
+    private static final String Engagment_add_URL = VolleyInstance.getBaseUrl() + "/engagments/engagments/";
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java:43]: I don't get it why you make it static? @KamilSupera @KonradMiskiewicz

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    @Override
+
+    /**
+     * Makes request to server and add engagment if success or shows
+     * error when failure.
+     */
+    public void onClick(View v)
+    {
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java:80]: Seems like this should go up

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+    private EditText advertisment_title_input;
+    private EditText advertisment_description_input;
+
+    private Spinner advertisment_city;
+
+    private Button sign_Advertisment;
+
+    private TextView ErrorTextView;
+
+    List<String> Cities ;
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: List<String> Cities; + why upperacase?

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+ */
+public class EngagmentAdd extends AppCompatActivity implements View.OnClickListener {
+    private EditText advertisment_title_input;
+    private EditText advertisment_description_input;
+
+    private Spinner advertisment_city;
+
+    private Button sign_Advertisment;
+
+    private TextView ErrorTextView;
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: Uppercase? it's field not a class

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+ * Class responsible for adding engagments
+ * Parsing data to JSON and sending to the server
+ */
+public class EngagmentAdd extends AppCompatActivity implements View.OnClickListener {
+    private EditText advertisment_title_input;
+    private EditText advertisment_description_input;
+
+    private Spinner advertisment_city;
+
+    private Button sign_Advertisment;
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: Decide between camel and snake case

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 15]
```diff
@@ -0,0 +1,146 @@
+package com.example.teamyuml.decentworkmobile.views;
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java:1]: I'm blind or here is no intent to this class. If you don't want make it in this feature add issue for it and if u want add it to Drawer Navigation menu.

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 16]
```diff
@@ -0,0 +1,146 @@
...
+
+    List<String> Cities ;
+
+    String selected_city;
+
+    private static final String Engagment_add_URL = VolleyInstance.getBaseUrl() + "/engagments/engagments/";
+    private static final String TAG = "Create Advertisment";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java:47]: RULE OF 30 IS CLOSE BABE. MY EYES ARE BLEEDING https://knowyourmeme.com/photos/1120230-reaction-images

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 17]
```diff
@@ -0,0 +1,146 @@
...
+     * Method is responsibility for adding params to engagment
+     * @return CreateJson object
+     */
+    public JSONObject addParams() {
+
+        String advertiser_title = advertisment_title_input.getText().toString();
+        String advertisment_description = advertisment_description_input.getText().toString();
+
+
+        CreateJson cj = new CreateJson();
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: I know i said to not include profession to check if this work but why it's not here? You can make condition if it's not empty add it to json.

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 18]
```diff
@@ -0,0 +1,146 @@
...
+        Cities.add("Łódź");
+        Cities.add("Poznań");
+        Cities.add("Wrocław");
+        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(EngagmentAdd.this,android.R.layout.simple_dropdown_item_1line,Cities);
+        arrayAdapter.setDropDownViewResource(android.R.layout.simple_dropdown_item_1line);
+        advertisment_city.setAdapter(arrayAdapter);
+        advertisment_city.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                selected_city = advertisment_city.getItemAtPosition(i).toString();
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: You don't have to make method or even listener for this `advertisment_city.getSelectedItem().toString();` in addParam should do the work

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 19]
```diff
@@ -0,0 +1,146 @@
...
+            }
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+            }
+        });
+        sign_Advertisment = findViewById(R.id.Advertiser_Add);
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: If you only use it in one method u don't have to make it global/class field

**Review by kaniak274 — COMMENTED** [review]
Yeah

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 20]
```diff
@@ -0,0 +1,146 @@
...
+        };
+        // Adding request to request queue
+        VolleyInstance.getInstance(this).addToRequestQueue(jsonObjectRequest, "Create Engagment");
+    }
+
+    /**
+     * Method is responsibility for adding params to engagment
+     * @return CreateJson object
+     */
+    public JSONObject addParams() {
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: I think this should be private, cause we don't want to use it in any other class

**Review by kaniak274 — COMMENTED** [review]
Decide one naming convention cause you one time use snake case, one time camel case, one time use start with uppercase

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 21]
```diff
@@ -0,0 +1,146 @@
...
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Class responsible for adding engagments
+ * Parsing data to JSON and sending to the server
+ */
+public class EngagmentAdd extends AppCompatActivity implements View.OnClickListener {
+    private EditText advertisment_title_input;
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: Why you call all of these `advertisment`? First this should be `advertisement`? I think announcement or notice are better words.

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 22]
```diff
@@ -0,0 +1,146 @@
...
+                    ErrorTextView.setText("pomyslnie dodano" + owner);
+                }
+                catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+        },  new Response.ErrorListener() {
+
+            @Override
+              public void onErrorResponse(VolleyError error) {
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java:97]: Wrong indent

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java [hunk 23]
```diff
@@ -0,0 +1,146 @@
...
+                try {
+                    String owner = response.getString("owner");
+                    ErrorTextView.setText("pomyslnie dodano" + owner);
+                }
+                catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+        },  new Response.ErrorListener() {
+
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: This enter decrease readability IMO

File: app/src/main/res/layout/activity_advertisement_form.xml [hunk 24]
```diff
@@ -0,0 +1,82 @@
...
+    <Button
+        android:id="@+id/Advertiser_Add"
+        android:layout_width="355dp"
+        android:layout_height="wrap_content"
+        android:text="Sign your advertisment"
+        tools:ignore="MissingConstraints"
+        tools:layout_editor_absoluteX="16dp"
+        tools:layout_editor_absoluteY="357dp" />
+
+    <TextView
```
Comment (kaniak274) [inline app/src/main/res/layout/activity_advertisement_form.xml]: For what is this TextView? Seems like it's for error text but i need to go 100 lines up to check it. Change id for something more descriptive or write comment.

File: app/src/main/res/layout/activity_advertisement_form.xml [hunk 25]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <TextView
```
Comment (kaniak274) [inline app/src/main/res/layout/activity_advertisement_form.xml]: Change id of those TextViews for something more descriptive.

File: app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java (hunk 18 above)
Comment (KonradMiskiewicz) [reply app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java]: I'm not sure about this. I need parameter of taken item and the convert to stream.

**Pull request merged**